notebooks/problem-sets/quantum_enigma_003.ipynb

```diff
@@ -0,0 +1,240 @@
+    "The enigma uses an addition modulo two like this one\n",
+    "\n",
+    "```code\n",
+    "   10\n",
+    "+  01\n",
+    "=  11\n",
+    "```\n",
+    "\n",
+    "Such addition has the interesting characteristic that the numbers can be interchanged to any order like this\n",
+    "\n",
+    "```code\n",
+    "   11\n",
+    "+  01\n",
+    "=  10\n",
+    "```\n",
+    "\n",
+    "Or\n",
+    "\n",
+    "```code\n",
+    "   11\n",
+    "+  10\n",
+    "=  01\n",
+    "```\n",
+    "\n",
+    "Meaning that adding any two of them gives the third one as an answer (this is true for any numbers). Playing with modulo two additions also have other interesting characteristics. In the enigma, adding the first number to the second is done by applying a `CNOT` between $q_4$ and $q_6$ (and $q_5$ and $q_7$). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-003-quiz-1\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "What is the value on $q_6$ after such an operation?\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. $q_6$ now has the answer to the modulo two addition between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. An extra qubit would be needed to have the answer to the modulo two addition between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "3. No addition has been performed between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "4. The `CNOT` does not permit to perform modulo two additions.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 2\n",
+    "\n",
+    "Can you write the circuit for a 4 by 4 square chess set until you calculate the position of the piece to turn?\n",
+    "\n",
+    "Check your answer by clicking the `Grade` button.\n",
+    "\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    Start by drawing a 4 by 4 chess board and number each square from 0 to 15 in decimal and binary numbers starting with the top row.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    The trick is now to add (modulo two) all the squares that end with a 1 and to proceed the same way with all squares that have a 1 on their second bit counting from right to left and so on using four extra squares.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 3</summary>\n",
```

Comment:
Is it possible to add a drawing?
I will provide one if so (we think it would greatly help the problem :) )

Reply:
Yep it is! Feel free to add it here and we can add it!

Reply:
It will come soon.

Reply:
![Echiquier 4x4](https://user-images.githubusercontent.com/64601645/206600834-91c2acb9-9727-43a8-a251-ca5fc2dbe226.png)